src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNSettings.java

```diff
@@ -422,22 +400,12 @@ public void setClusterService(ClusterService clusterService) {
     static class SpaceTypeValidator implements Setting.Validator<String> {
 
         @Override public void validate(String value) {
-            if (value == null || !SpaceTypes.contains(value.toLowerCase())){
+            if (value == null || !SpaceType.getValues().contains(value.toLowerCase())){
                 throw new InvalidParameterException(String.format("Unsupported space type: %s", value));
```

Comment:
if this is user facing message, can we add what are supported types as well? This will give more information to the user. What do you think?

Reply:
Yes, I can add that.